frontend/app/utils/route-utils.ts

```diff
@@ -4,9 +4,29 @@ import { z } from 'zod';
 
 const pageTitleDataSchema = z.object({ pageTitle: z.string() });
 
+export type PageTitleDataSchema = z.infer<typeof pageTitleDataSchema>;
+
 export function usePageTitle() {
   return useMatches()
     .map(({ data }) => pageTitleDataSchema.safeParse(data))
     .map((result) => (result.success ? result.data.pageTitle : undefined))
     .reduce((last, curr) => curr ?? last, undefined);
 }
+
+const breadcrumbDataSchema = z.object({
+  breadcrumb: z.array(
+    z.object({
+      label: z.string(),
+      to: z.string().optional(),
+    }),
+  ),
+});
+
+export type BreadcrumbDataSchema = z.infer<typeof breadcrumbDataSchema>;
+
+export function useBreadcrumb() {
```

Comment:
**Nitpick:** should this be `useBreadcrumbs()`?

Reply:
🤷

The correct term is "breadcrumb" without an "s." In the context of a website, a breadcrumb is a navigational aid that helps users understand their location on a site and easily navigate back to previous pages. It's called a breadcrumb trail, and the term is singular in this context. So, you would refer to it as "breadcrumb" when discussing website navigation.

🤷

Reply:
I guess you're right.

> If your React hook returns an array that may contain zero or multiple breadcrumb items, you might want to use the plural form "breadcrumbs" when referring to the array as a whole. This is because the array could potentially represent multiple instances of the breadcrumb.